kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt

```diff
@@ -6,58 +6,83 @@ import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
-      // use a custom style to create a dictionary renderer and set it to the feature layer renderer
-      renderer = DictionaryRenderer(
-        DictionarySymbolStyle.createFromFile(
-          getExternalFilesDir(null)?.path + getString(R.string.restaurant_stylx_path)
-        )
-      )
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // authentication with an API key or named user is required to access basemaps and other
+        // location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        // create a feature layer from a service feature table
+        val featureTable =
+            ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/Redlands_Restaurants/FeatureServer/0")
+        val featureLayer = FeatureLayer(featureTable)
+
+        // create a new map with a streets basemap and set it to the map view
+        mapView.map = ArcGISMap(BasemapStyle.ARCGIS_STREETS).apply {
+            // add the the feature layer to the map's operational layers
+            operationalLayers.add(featureLayer)
+            // set the initial viewpoint to the Esri Redlands campus
+            initialViewpoint = Viewpoint(34.0574, -117.1963, 5000.0)
+        }
+
+        // create a dictionary symbol style from the stylx file
+        val dictionarySymbolStyleFromFile =
+            DictionarySymbolStyle.createFromFile(getExternalFilesDir(null)?.path + "/Restaurant.stylx")
+        // create a new dictionary renderer from the dictionary symbol style
+        val dictionaryRendererFromFile = DictionaryRenderer(dictionarySymbolStyleFromFile)
+
+        // on style file click
+        styleFileRadioButton.setOnClickListener {
+            // set the feature layer renderer to the dictionary renderer from local stylx file
+            featureLayer.renderer = dictionaryRendererFromFile
+        }
+        // set the initial state to use the dictionary renderer from local stylx file
+        styleFileRadioButton.performClick()
+
+        // create a portal item using the portal and the item id of the dictionary web style
+        val portal = Portal("https://arcgisruntime.maps.arcgis.com")
+        val portalItem = PortalItem(portal, "adee951477014ec68d7cf0ea0579c800")
+        // map the input fields in the feature layer to the dictionary symbol style's expected fields for symbols and text
+        val fieldMap: HashMap<String, String> = HashMap()
+        fieldMap["healthgrade"] = "Inspection"
+        // create a new dictionary symbol style from the web style in the portal item
+        val dictionarySymbolStyleFromPortal = DictionarySymbolStyle(portalItem)
+        // create a new dictionary renderer from the dictionary symbol style
+        val dictionaryRendererFromPortal = DictionaryRenderer(dictionarySymbolStyleFromPortal)
```

Comment:
```suggestion
        val dictionaryRendererFromPortal = DictionaryRenderer(dictionarySymbolStyleFromPortal, fieldMap, HashMap())
```
The fieldMap hasn't been applied... 😅 

Also, the screenshot needs update.